sdk/storage/Azure.Storage.Common/src/StorageSharedKeyPipelinePolicy.cs

```diff
@@ -106,7 +106,7 @@ private static string BuildCanonicalizedHeaders(HttpMessage message)
                     sb.Append('\n');
                 }
                 message.Request.Headers.TryGetValue(headerName, out var value);
-                sb.Append(headerName).Append(':').Append(value);
+                sb.Append(headerName.ToLowerInvariant()).Append(':').Append(value);
```

Comment:
Can we move this up into the `Select` on line 98 and avoid doing the `ToLowerInvariant` twice here and in `OrderBy` on line 101?